sdk/communication/azure-communication-chat/src/test/java/com/azure/communication/chat/ChatThreadAsyncClientTest.java

```diff
@@ -64,158 +67,219 @@ protected void afterTest() {
     public void canSendThenGetMessage() {
+        // Arrange
         SendChatMessageOptions messageRequest = ChatOptionsProvider.sendMessageOptions();
-
-        SendChatMessageResult response = chatThreadClient.sendMessage(messageRequest).block();
-
-        ChatMessage message = chatThreadClient.getMessage(response.getId()).block();
-        assertEquals(message.getContent(), messageRequest.getContent());
-        assertEquals(message.getPriority(), messageRequest.getPriority());
-        assertEquals(message.getSenderDisplayName(), messageRequest.getSenderDisplayName());
+        
+        // Action & Assert
+        StepVerifier.create(chatThreadClient.sendMessage(messageRequest))
+            .assertNext(response -> {
+                StepVerifier.create(chatThreadClient.getMessage(response.getId()))
+                    .assertNext(message -> {
+                        assertEquals(message.getContent(), messageRequest.getContent());
+                        assertEquals(message.getPriority(), messageRequest.getPriority());
+                        assertEquals(message.getSenderDisplayName(), messageRequest.getSenderDisplayName());
+                    })
+                    .verifyComplete();
+            })
+            .verifyComplete();
     }
 
     @Test
     public void canSendThenGetMessageWithResponse() {
+        // Arrange
         SendChatMessageOptions messageRequest = ChatOptionsProvider.sendMessageOptions();
 
-        SendChatMessageResult response = chatThreadClient.sendMessageWithResponse(messageRequest).block().getValue();
-
-        ChatMessage message = chatThreadClient.getMessageWithResponse(response.getId()).block().getValue();
-        assertEquals(message.getContent(), messageRequest.getContent());
-        assertEquals(message.getPriority(), messageRequest.getPriority());
-        assertEquals(message.getSenderDisplayName(), messageRequest.getSenderDisplayName());
+        // Action & Assert
+        StepVerifier.create(chatThreadClient.sendMessageWithResponse(messageRequest))
+            .assertNext(sendResponse -> {
+                StepVerifier.create(chatThreadClient.getMessageWithResponse(sendResponse.getValue().getId()))
+                    .assertNext(getResponse -> {
+                        ChatMessage message = getResponse.getValue();
+                        assertEquals(message.getContent(), messageRequest.getContent());
+                        assertEquals(message.getPriority(), messageRequest.getPriority());
+                        assertEquals(message.getSenderDisplayName(), messageRequest.getSenderDisplayName());
+                    })
+                    .verifyComplete();
+            })
+            .verifyComplete();
     }
 
     @Test
     public void canDeleteExistingMessage() {
+        // Arrange
         SendChatMessageOptions messageRequest = ChatOptionsProvider.sendMessageOptions();
-
         SendChatMessageResult response = chatThreadClient.sendMessage(messageRequest).block();
```

Comment:
Same comment here, need to get rid of this .block() too

Reply:
Done